Cubical/Functions/Embedding.agda

```diff
@@ -186,3 +199,137 @@ Embedding-into-isSet→isSet (f , hasPropFibers-f) isSet-B x y p q =
+    embedIsProp (isEquiv→isEmbedding (snd fiberChar))
+      (isPropΣ (isEmbedding→hasPropFibers (liftEmbedding _ _) A)
+               λ _ → isEquiv→hasPropFibers (snd (invEquiv (preCompEquiv LiftEquiv))) _)
+    where
+    fiberChar : fiber liftFibration (A , f)
+              ≃ (Σ[ (E , eq) ∈ fiber L A ] fiber (_∘ lower) (transport⁻ (λ i → eq i → B) f))
+    fiberChar =
+        fiber liftFibration (A , f)
+      ≃⟨ Σ-cong-equiv-snd (λ _ → invEquiv ΣPath≃PathΣ) ⟩
+        (Σ[ (E , g) ∈ Fibration B ℓ₁ ] Σ[ eq ∈ (L E ≡ A) ] PathP (λ i → eq i → B) (g ∘ lower) f)
+      ≃⟨ boringSwap ⟩
+        (Σ[ (E , eq) ∈ fiber L A ] Σ[ g ∈ (E → B) ] PathP (λ i → eq i → B) (g ∘ lower) f)
+      ≃⟨ Σ-cong-equiv-snd (λ _ → Σ-cong-equiv-snd λ _ → transportEquiv (PathP≡Path⁻ _ _ _)) ⟩
+        (Σ[ (E , eq) ∈ fiber L A ] fiber (_∘ lower) (transport⁻ (λ i → eq i → B) f))
+      ■ where
+      boringSwap : _
+      boringSwap = isoToEquiv (iso (λ ((E , g) , (eq , p)) → ((E , eq) , (g , p)))
+                                   (λ ((E , g) , (eq , p)) → ((E , eq) , (g , p))) (λ _ → refl) (λ _ → refl))
+
+  isEmbeddingLiftFibration : isEmbedding liftFibration
+  isEmbeddingLiftFibration = hasPropFibers→isEmbedding hasPropFibersLiftFibration
+
+  -- and finish off
+  module _ (f g : Fibration B ℓ₁) where
+    open Lifted (liftFibration f) (liftFibration g)
+    f≃g : Type (ℓ-max ℓ ℓ₁)
+    f≃g = ∀ b → fiber (f .snd) b ≃ fiber (g .snd) b
+
+    FibrationIP : f≃g ≃ (f ≡ g)
+    FibrationIP =
+      f≃g  ≃⟨ equivΠCod (λ b → equivComp (Σ-cong-equiv-fst LiftEquiv)
+                                          (Σ-cong-equiv-fst LiftEquiv)) ⟩
+      f≃g′ ≃⟨ Fibration′IP ⟩
+      (liftFibration f ≡ liftFibration g) ≃⟨ invEquiv (_ , isEmbeddingLiftFibration _ _) ⟩
+      (f ≡ g) ■
+open FibrationIdentityPrinciple renaming (f≃g to _≃Fib_) using (FibrationIP) public
+
+EmbeddingStr : Type ℓ₁ → Type ℓ → Type (ℓ-max ℓ ℓ₁)
+EmbeddingStr B = AxiomsStructure (λ A → A → B) (λ _ → isEmbedding)
+
+Embedding : (B : Type ℓ₁) → (ℓ : Level) → Type (ℓ-max ℓ₁ (ℓ-suc ℓ))
+Embedding B ℓ = Σ[ A ∈ Type ℓ ] EmbeddingStr B A
+
+module EmbeddingIdentityPrinciple {B : Type ℓ} {ℓ₁} (f g : Embedding B ℓ₁) where
+  module _ where
+    open Σ f renaming (fst to F) public
+    open Σ g renaming (fst to G) public
+    open Σ (f .snd) renaming (fst to ffun; snd to isEmbF) public
+    open Σ (g .snd) renaming (fst to gfun; snd to isEmbG) public
+  f≃g : Type _
+  f≃g = (∀ b → fiber ffun b → fiber gfun b) ×
+         (∀ b → fiber gfun b → fiber ffun b)
+  toFibr : Embedding B ℓ₁ → Fibration B ℓ₁
+  toFibr (A , (f , _)) = (A , f)
+
+  isEmbeddingToFibr : isEmbedding toFibr
+  isEmbeddingToFibr w x = fullEquiv .snd where
+    -- carefully managed such that (cong toFibr) is the equivalence
+    fullEquiv : (w ≡ x) ≃ (toFibr w ≡ toFibr x)
+    fullEquiv = compEquiv (congEquiv (invEquiv Σ-assoc-≃)) (invEquiv (Σ≡PropEquiv (λ _ → isEmbeddingIsProp)))
+
+  EmbeddingIP : f≃g ≃ (f ≡ g)
```

Comment:
Hmm, could this proof use the SIP machinery that's already in the library?

Reply:
Not sure, but it didn't seem so. SIP would give an equivalence `F ≃ G` and some other stuff, but that equivalence is not directly found in `f≃g`.

Reply:
Hmm, maybe not... What does @ecavallo think?

Reply:
I agree, I don't think the SIP will help here.

Reply:
Merci